Framework/Parallel/inc/MantidParallel/IO/NXEventDataLoader.h

```diff
@@ -0,0 +1,237 @@
+                    std::vector<std::string> bankNames);
+
+  std::unique_ptr<AbstractEventDataPartitioner<TimeOffsetType>>
+  setBankIndex(const size_t bank) override;
+
+  void readEventID(int32_t *event_id, size_t start,
+                   size_t count) const override;
+  void readEventTimeOffset(TimeOffsetType *event_time_offset, size_t start,
+                           size_t count) const override;
+  std::string readEventTimeOffsetUnit() const override;
+
+private:
+  const int m_numWorkers;
+  const H5::Group m_root;
+  H5::Group m_group;
+  const std::vector<std::string> m_bankNames;
+  H5::DataSet m_id;
+  H5::DataSet m_time_offset;
+};
+
+namespace detail {
+/// Read complete data set from group and return the contents as a vector.
+template <class T>
+std::vector<T> read(const H5::Group &group, const std::string &dataSetName) {
+  H5::DataSet dataSet = group.openDataSet(dataSetName);
+  H5::DataType dataType = dataSet.getDataType();
+  H5::DataSpace dataSpace = dataSet.getSpace();
+  std::vector<T> result;
+  result.resize(dataSpace.getSelectNpoints());
+  dataSet.read(result.data(), dataType);
+  return result;
+}
+
+/** Read subset of data set and write the result into buffer.
+ *
+ * The subset is given by a start index and a count. */
+template <class T>
+void read(T *buffer, const H5::DataSet &dataSet, size_t start, size_t count) {
+  auto hstart = static_cast<hsize_t>(start);
+  auto hcount = static_cast<hsize_t>(count);
+  H5::DataType dataType = dataSet.getDataType();
+  H5::DataSpace dataSpace = dataSet.getSpace();
+  if ((static_cast<int64_t>(dataSpace.getSelectNpoints()) -
+       static_cast<int64_t>(hstart)) <= 0)
+    throw std::out_of_range("Start index is beyond end of file");
+  if (hcount > dataSpace.getSelectNpoints() - hstart)
+    throw std::out_of_range("End index is beyond end of file");
+  dataSpace.selectHyperslab(H5S_SELECT_SET, &hcount, &hstart);
+  H5::DataSpace memSpace(1, &hcount);
+  dataSet.read(buffer, dataType, memSpace, dataSpace);
+}
+
+/** Read subset of data set from group and write the result into buffer.
+ *
+ * The subset is given by a start index and a count. */
+template <class T>
+void read(T *buffer, const H5::Group &group, const std::string &dataSetName,
+          size_t start, size_t count) {
+  H5::DataSet dataSet = group.openDataSet(dataSetName);
+  read(buffer, dataSet, start, count);
+}
+
+std::string readAttribute(const H5::DataSet &dataSet,
+                          const std::string &attributeName) {
+  const auto &attr = dataSet.openAttribute(attributeName);
+  std::string value;
+  attr.read(attr.getDataType(), value);
+  return value;
+}
+
+template <class TimeOffsetType, class IndexType, class TimeZeroType>
+std::unique_ptr<AbstractEventDataPartitioner<TimeOffsetType>>
+makeEventDataPartitioner(const H5::Group &group, const int numWorkers) {
+  const auto timeZero = group.openDataSet("event_time_zero");
+  int64_t time_zero_offset{0};
+  if (timeZero.attrExists("offset")) {
```

Comment:
I get an error here when building on Ubuntu 14.04. I guess this is a HDF5 version issue?